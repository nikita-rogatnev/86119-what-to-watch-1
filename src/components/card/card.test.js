import React from 'react';
import renderer from 'react-test-renderer';

import Card from './card';

const mockData = {
  "id": 1,
  "name": `Seven Years in Tibet`,
  "previewImage": `https://es31-server.appspot.com/wtw/static/film/preview/seven-years-in-tibet.jpg`,
  "previewVideoLink": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
};

it(`Card renders correctly`, () => {
  const tree = renderer
    .create(<Card
      key={mockData.id}
      id={mockData.id}
      previewImage={mockData.previewImage}
      previewVideoLink={mockData.previewVideoLink}
      showButton={false}
    />).toJSON();

  expect(tree).toMatchSnapshot();
});
