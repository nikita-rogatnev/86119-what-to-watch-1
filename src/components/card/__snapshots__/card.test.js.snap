// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  
  <div
    className="small-movie-card__image"
  >
    <img
      alt="Matrix"
      height="175"
      src="https://es31-server.appspot.com/wtw/static/film/preview/matrix.jpg"
      width="280"
    />
  </div>
  <h3
    className="small-movie-card__title"
  >
    Action
    <br />
    <a
      className="small-movie-card__link"
      href="/films/1"
      onClick={[Function]}
    >
      Matrix
    </a>
  </h3>
</article>
`;
